### PR TITLE
feat: creat crds from folder

### PR DESCRIPTION
**Pull request by nicufk**
## Pull request description 
Closing https://github.com/kubeshop/testkube/issues/3125

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/test/detector/curltest.go [hunk 1]
```diff
@@ -31,6 +32,14 @@ func (d CurlTestAdapter) Is(options apiClient.UpsertTestOptions) (name string, o
 	return
 }
 
+// IsWithPath detects based on upsert test options what kind of test it is
+func (d CurlTestAdapter) IsWithPath(path string, options apiClient.UpsertTestOptions) (name string, ok bool) {
```
Comment (vsukhin) [inline pkg/test/detector/curltest.go:36]: should we move it to executor folder and create detector, validator packages? we can register them at app start

File: pkg/test/detector/maven.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"strings"
+
+	apiClient "github.com/kubeshop/testkube/pkg/api/v1/client"
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+)
+
+// has multiple files and a pom.xml file
+
+// MavenAdapter is detector adapter for Maven test
+type MavenAdapter struct{}
```
Comment (vsukhin) [inline pkg/test/detector/maven.go:13]: Sounds like,we have multiple copies of this behaivot.Might be to add DefaultAdapter with a few params?

File: cmd/kubectl-testkube/commands/common/repository.go [hunk 3]
```diff
@@ -8,6 +8,9 @@ import (
 
 // NewRepositoryFromFlags creates repository from command flags
 func NewRepositoryFromFlags(cmd *cobra.Command) (repository *testkube.Repository, err error) {
+	if cmd.Flag("git-uri") == nil {
```
Comment (vsukhin) [inline cmd/kubectl-testkube/commands/common/repository.go:11]: why did we add it? we callit from somewhere without this flag?

File: cmd/kubectl-testkube/commands/crds/tests_crds.go [hunk 4]
```diff
@@ -1,192 +1,252 @@
...
 
-			err = filepath.Walk(dir, func(path string, info fs.FileInfo, err error) error {
-				if err != nil {
-					return nil
-				}
-
-				if filepath.Ext(path) != ".json" {
-					return nil
-				}
+	cmd.Flags().StringVarP(&executorType, "type", "t", "", "test type, if it is specified it will generate only tests of this type")
```
Comment (vsukhin) [inline cmd/kubectl-testkube/commands/crds/tests_crds.go:112]: might be we need tohave a common part instead of copying

**Review by vsukhin — COMMENTED** [review]
some comments

File: cmd/kubectl-testkube/commands/common/repository.go (hunk 3 above)
Comment (nicufk) [reply cmd/kubectl-testkube/commands/common/repository.go:11]: yes, we call it from generate cmd, and it doesn't have this param, maybe we can put a check where this func is called.

File: pkg/test/detector/maven.go (hunk 2 above)
Comment (nicufk) [reply pkg/test/detector/maven.go:13]: This is what I am working on next

File: pkg/test/detector/curltest.go (hunk 1 above)
Comment (nicufk) [reply pkg/test/detector/curltest.go:36]: Yes, I think it is a good idea to move them to executors.

**Comment by gberche-orange** [comment]
@nicufk maybe part of the documentation updates proposed into https://github.com/kubeshop/testkube/pull/3534 should be adapted to `testkube create with flag --crd-only` ?

File: docs/docs/reference/cli/testkube_create_test.md [hunk 5]
```diff
@@ -66,7 +68,7 @@ testkube create test [flags]
   -c, --client string      client used for connecting to Testkube API one of proxy|direct (default "proxy")
       --crd-only           generate only crd
       --namespace string   Kubernetes namespace, default value read from config if set (default "testkube")
-      --oauth-enabled      enable oauth
+      --oauth-enabled      enable oauth (default true)
```
Comment (vsukhin) [inline docs/docs/reference/cli/testkube_create_test.md:71]: should be false

File: docs/docs/reference/cli/testkube_create_test.md [hunk 6]
```diff
@@ -45,12 +46,13 @@ testkube create test [flags]
...
       --schedule string                            test schedule in a cronjob form: * * * * *
       --scraper-template string                    scraper template file path for extensions to scraper template
+      --secret-env stringToString                  secret envs in a form of secret_key1=secret_name1 passed to executor (default [])
   -s, --secret-variable stringToString             secret variable key value pair: --secret-variable key1=value1 (default [])
       --secret-variable-reference stringToString   secret variable references in a form name1=secret_name1=secret_key1 (default [])
       --source string                              source name - will be used together with content parameters
       --test-content-type string                   content type of test one of string|file-uri|git
       --timeout int                                duration in seconds for test to timeout. 0 disables timeout.
-  -t, --type string                                test type (defaults to postman/collection)
+  -t, --type string                                test type
```
Comment (vsukhin) [inline docs/docs/reference/cli/testkube_create_test.md:55]: should be postman?

File: cmd/kubectl-testkube/commands/crds/tests_crds.go [hunk 7]
```diff
@@ -1,192 +1,181 @@
...
 )
 
 // NewCRDTestsCmd is command to generate test CRDs
 func NewCRDTestsCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "tests-crds <manifestDirectory>",
-		Short: "Generate tests CRD file based on directory",
-		Long:  `Generate tests manifest based on directory (e.g. for ArgoCD sync based on tests files)`,
-		Args:  validator.ManifestsDirectory,
+		Use:     "tests-crds <manifestDirectory>",
```
Comment (vsukhin) [inline cmd/kubectl-testkube/commands/crds/tests_crds.go:31]: should we markit as deprected?

**Review by vsukhin — APPROVED** [review]
looks good, main concern is backward compatibility and marking old command as deprecated

File: docs/docs/reference/cli/testkube_create_test.md (hunk 5 above)
Comment (nicufk) [inline docs/docs/reference/cli/testkube_create_test.md:71]:
```suggestion
      --oauth-enabled      enable oauth
```

File: docs/docs/reference/cli/testkube_create_test.md (hunk 6 above)
Comment (nicufk) [reply docs/docs/reference/cli/testkube_create_test.md:55]: if it cannot detect type it should give an error

**Comment by nicufk** [comment]
> @nicufk maybe part of the documentation updates proposed into #3534 should be adapted to `testkube create with flag --crd-only` ?

@gberche-orange as I mentioned above `testkube generate` will be used to generate the crds from an entire folder and as you can see from the changes in this PR it is using `create test --crd-only` so every feature that is added to create will be reflected in the generate as well

**Comment by gberche-orange** [comment]
> > @nicufk maybe part of the documentation updates proposed into #3534 should be adapted to `testkube create with flag --crd-only` ?
> 
> @gberche-orange as I mentioned above `testkube generate` will be used to generate the crds from an entire folder and as you can see from the changes in this PR it is using `create test --crd-only` so every feature that is added to create will be reflected in the generate as well

As I was reading the linked issue  https://github.com/kubeshop/testkube/issues/3125 which mentions!
> Ideally we would want to only have testkube create test --crds-only and deprecate testkube generate test-crds once we have feature parity.

I was worrying that the `testkube_generate_tests-crds.md` would be removed once the `testkube generate test-crds` command would be deprecated, along with the corresponding doc 

https://github.com/kubeshop/testkube/blob/f8789afef4a17327bbe2c02f3ef69bc7346c968a/docs/docs/reference/cli/testkube_generate_tests-crds.md?plain=1#L9-L14

**Pull request merged**